From e13752f5ac89a9bb7046597a4b024216ae8fabef Mon Sep 17 00:00:00 2001
From: Linjie Fu <linjie.fu@intel.com>
Date: Fri, 27 Mar 2020 14:01:44 +0800
Subject: [PATCH 25/81] lavc/vaapi_encode_h265: fix
 max_transform_hierarchy_depth_inter/intra

Set the max_transform_hierarchy_depth_inter/intra to 2 by default
based on the Programmer's Reference Manuals (PRM) in [1].

Intel Encoder only supports 2 levels of quad-tree. That is:
    - max_transform_hierarchy_depth_inter/intra <= 2.

[1] <https://01.org/sites/default/files/documentation/intel-gfx-prm-osrc-kbl-vol10-hevc.pdf>

Signed-off-by: Linjie Fu <linjie.fu@intel.com>
---
 libavcodec/vaapi_encode_h265.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/libavcodec/vaapi_encode_h265.c b/libavcodec/vaapi_encode_h265.c
index a04c1ccd02..87dbd7d50e 100644
--- a/libavcodec/vaapi_encode_h265.c
+++ b/libavcodec/vaapi_encode_h265.c
@@ -432,8 +432,9 @@ static int vaapi_encode_h265_init_sequence_params(AVCodecContext *avctx)
     sps->log2_min_luma_transform_block_size_minus2   = 0;
     sps->log2_diff_max_min_luma_transform_block_size = 3;
     // Full transform hierarchy allowed (2-5).
-    sps->max_transform_hierarchy_depth_inter = 3;
-    sps->max_transform_hierarchy_depth_intra = 3;
+    // Default to 2 based on Programmer's Reference Manuals of Intel graphics
+    sps->max_transform_hierarchy_depth_inter = 2;
+    sps->max_transform_hierarchy_depth_intra = 2;
     // AMP works.
     sps->amp_enabled_flag = 1;
     // SAO and temporal MVP do not work.
-- 
2.25.1

